src/main/java/hu/wurfel/new_school_reference/teacher/TeacherDto.java

```diff
@@ -14,8 +14,8 @@ public class TeacherDto extends BaseDto {
 	private long cardNumber;
 
 	@Override
-	public boolean hasId() {
-		return id != null;
+	public boolean hasValidId() {
+		return id != null && id >= 1;
```

Comment:
```suggestion
		return id != null && id > 0;
```
Bár pontosan ugyanazt jelenti, compaktabb és kevésbé siklik el felette az ember, amikor átnézi, mert a logikai operátor 1 karakter és nem 2. Ez néha elég sokat tud számítani. 😄